product_docs/docs/biganimal/release/migration/dha_bulk_migration.mdx

```diff
@@ -0,0 +1,389 @@
+
+TARGET_DB_URL=jdbc:postgresql://<target-host>:<target-port>/<target-dbname>
+TARGET_DB_USER=<target-user>
+TARGET_DB_PASSWORD=<target-password>
+```
+
+Edit the relevant values into the settings:
+
+Ensure that the configuration file is owned by the user you intend to run the data transfer as and read-write only for its owner.
+
+Now, select sets of tables in the source database that should be be transferred together, ideally grouping them for redundancy in case of failure.
+
+```
+nohup /usr/edb/migrationtoolkit/bin/runMTK.sh -sourcedbtype postgres -targetdbtype postgres  -loaderCount 1  -tableLoaderLimit 1  -fetchSize 4000  -parallelLoadRowLimit 1000  -truncLoad -dataOnly -tables <tablename1>,<tablename2>,... <schemaname> > mtk.log
+```
+
+This command uses the `-truncLoad` option and will drop indexes and constraints before the data is loaded, then recreate them after the loading has completed.
+
+You can run multiple instances of this command in parallel; add an `&` to the end of the command. Ensure that you write the output from each to different files (e.g mtk_1.log, mtk_2.log). 
+
+For example:
+
+```
+nohup /usr/edb/migrationtoolkit/bin/runMTK.sh -sourcedbtype postgres -targetdbtype postgres  -loaderCount 1  -tableLoaderLimit 1  -fetchSize 4000  -parallelLoadRowLimit 1000  -truncLoad -dataOnly -tables warehouse,district,item,new_order,orders,history public >mtk_1.log &
+
+nohup /usr/edb/migrationtoolkit/bin/runMTK.sh -sourcedbtype postgres -targetdbtype postgres -loaderCount 1  -tableLoaderLimit 1  -fetchSize 4000  -parallelLoadRowLimit 1000  -truncLoad -dataOnly -tables customer public >mtk_2.log &
+
+nohup /usr/edb/migrationtoolkit/bin/runMTK.sh -sourcedbtype postgres -targetdbtype postgres  -loaderCount 1  -tableLoaderLimit 1  -fetchSize 4000  -parallelLoadRowLimit 1000  -truncLoad -dataOnly -tables order_line public >mtk_3.log &
+
+nohup /usr/edb/migrationtoolkit/bin/runMTK.sh -sourcedbtype postgres -targetdbtype postgres  -loaderCount 1  -tableLoaderLimit 1  -fetchSize 4000  -parallelLoadRowLimit 1000  -truncLoad -dataOnly -tables stock public >mtk_4.log &
+```
+
+This sets up four processes, each transferring a particular table or sets of tables as a background process. 
+
+While this is running, monitor the lag. Log into the destination node with psql and monitor lag with:
+
+```sql
+SELECT NOW(); SELECT pg_size_pretty( pg_database_size('bdrdb') ); SELECT * FROM bdr.node_replication_rates;
+```
+
+Once the lag has been consumed, return to the shell. You can now use `tail` to monitor the progress of the data transfer by following the log files of each process:
+
+```
+tail -f mtk_1.log
+```
+
+### Transferring the post-data
+
+Make sure there is no replication lag across the entire cluster before proceeding with post-data.
+
+Now dump the post-data section of the source database:
+
+```
+pg_dump -Fd -f postdata --section=pre-data -h <source-hostname> -p <source-port> -U <source-user> <source-dbname>
```

Comment:
@sszhao1020 you mentioned you'd add a `-t <table>` here, but I guess that's optional in case you'd like to execute post-data separately across tables?

Reply:
Yes, correct.